test/unit/org/apache/cassandra/cql3/validation/entities/SecondaryIndexTest.java

```diff
@@ -70,6 +72,12 @@
 public class SecondaryIndexTest extends CQLTester
 {
     public static final int TOO_BIG = 1024 * 65;
+    
+    @BeforeClass
+    public static void setDefaultSecondaryIndex()
+    {
+        DatabaseDescriptor.setDefaultSecondaryIndex(CassandraIndex.NAME);
```

Comment:
this test do not cover that when we do not set defaultSecondaryIndex.

Reply:
There were two options to make sure this test works as intended independent of what the YAML says. One is this, just making sure the legacy 2i is the default, and none of the tests themselves need to change their index creation DDL. The other would be changing all the index creation DDL. I chose the former to keep the diff smaller.

`shouldCreateCassandraIndexExplicitly` though...that could change, or a new test be added to check explicit legacy 2i creation when it isn't the default.

Reply:
Added a new case... `shouldCreateCassandraIndexWhenNotDefault`